RFCs/Concepts_Redesign.md

```diff
@@ -0,0 +1,147 @@
+# Concepts redesign
+
+
+## Abstract
+
+This RFC proposes a simpler concept syntax, one that cuts right to the point of
+what a concept specifies.
+
+
+## Motivation
+
+The current concepts syntax is difficult to grok as it requires mentally
+evaluating Nim code. The concepts implementation is also currently very buggy
+and would require significant effort to make reliable.
+
+A simpler syntax will solve implementation problems and aid adoption of the concepts.
+It should also make it simpler to implement a "runtime concept".
+
+
+Everyone who wants define interfaces/concepts/traits will benefit from this.
+
+
+## Description
+
+The essence of this proposal is to overhaul concepts completely, instead of
+requiring a concepts specification to be defined in terms of code that is evaluated
+at compile-time I propose we settle on specifying concrete procedures/iterators/methods
+that need to be defined on the concept that we are specifying.
+
+My proposal very much takes inspiration from implementations of traits/interfaces
+in other languages. That isn't a bad thing, in fact it's a very good thing.
+
+A concrete specification of which procedures need to be defined on a type makes
+understanding a concept far easier to developers. The biggest problem with the
+current concepts is that they need to be evaluated, evaluating a concept when
+trying to determine why it doesn't match your type can be really time consuming
+and frustrating. What this RFC proposes will make it significantly easier to
+understand what a concept specifies and to figure out why a type doesn't
+satisfy its spec.
+
+The downsides to this proposal are that what can be specified is diminished,
+that said I don't believe this is significant and I go through examples of this
+below.
```

Comment:
This RFC should not remove the "old" concepts, IMO.

Reply:
Not immediately, but eventually it should.